app/src/main/java/co/smartreceipts/android/distance/editor/DistanceAutoCompletePresenter.kt

```
@@ -0,0 +1,58 @@
+package co.smartreceipts.android.distance.editor
+
+import co.smartreceipts.android.autocomplete.distance.DistanceAutoCompleteField
+import co.smartreceipts.android.model.factory.DistanceBuilderFactory
+import co.smartreceipts.android.widget.viper.BaseViperPresenter
+import co.smartreceipts.core.di.scopes.FragmentScope
+import javax.inject.Inject
+
+@FragmentScope
+class DistanceAutoCompletePresenter @Inject constructor(
+    view: DistanceCreateEditView, interactor: DistanceCreateEditInteractor
+) :
+    BaseViperPresenter<DistanceCreateEditView, DistanceCreateEditInteractor>(view, interactor) {
+
+    override fun subscribe() {
+
+        compositeDisposable.add(view.hideAutoCompleteVisibilityClick
+                .flatMap { t ->
+                    when (t.type) {
+                        DistanceAutoCompleteField.Location -> interactor.updateDistance(t.item, DistanceBuilderFactory(t.item)
+                                .setLocationHiddenFromAutoComplete(true)
+                                .build())
+                        DistanceAutoCompleteField.Comment -> interactor.updateDistance(t.item, DistanceBuilderFactory(t.item)
+                                .setCommentHiddenFromAutoComplete(true)
+                                .build())
+                        else -> interactor.updateDistance(t.item, t.item)
```

I think we can do `throw UnsupportedOperationException(...)` here, since we only support these two types, correct?

---

Also to double check. Do we need the else block? Or can we delete it there?